Quarkus dev mode does not work

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Developer experience is broken. This affects:
* Running [`IDELauncher`](https://github.com/keycloak/keycloak/blob/36b01cbea093d38c495c9e0f2ef66feb3a3f9e41/quarkus/server/src/main/java/org/keycloak/quarkus/_private/IDELauncher.java) from the IDE.
* Running `mvn -f server/pom.xml compile quarkus:dev -Dquarkus.args="start-dev"`.

This is caused by https://github.com/keycloak/keycloak/pull/34309.

Workaround is to use either the [`Keycloak` class](https://github.com/keycloak/keycloak/blob/36b01cbea093d38c495c9e0f2ef66feb3a3f9e41/quarkus/tests/junit5/src/main/java/org/keycloak/Keycloak.java) in junit5 module or [`KeycloakServer` class](https://github.com/keycloak/keycloak/blob/36b01cbea093d38c495c9e0f2ef66feb3a3f9e41/testsuite/utils/src/main/java/org/keycloak/testsuite/KeycloakServer.java) in utils.

### Version

main,release/26.0

### Regression

- [X] The issue is a regression

### Expected behavior

Server starts.

### Actual behavior

```
ERROR [io.quarkus.deployment.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devui.deployment.BuildTimeContentProcessor#createIndexHtmlTemplate threw an exception: java.lang.NoClassDefFoundError: io/mvnpm/importmap/Aggregator
	at io.quarkus.devui.deployment.BuildTimeContentProcessor.createIndexHtmlTemplate(BuildTimeContentProcessor.java:344)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: java.lang.ClassNotFoundException: io.mvnpm.importmap.Aggregator
	at io.quarkus.launcher.RuntimeLaunchClassLoader.findClass(RuntimeLaunchClassLoader.java:25)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:533)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:481)
	... 10 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:354)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:272)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:62)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:91)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:430)
	at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:71)
	at io.quarkus.deployment.dev.IDEDevModeMain.accept(IDEDevModeMain.java:28)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
	at io.quarkus.bootstrap.IDELauncherImpl.launch(IDELauncherImpl.java:93)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.launcher.QuarkusLauncher.launch(QuarkusLauncher.java:56)
	at io.quarkus.runtime.Quarkus.launchFromIDE(Quarkus.java:101)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:88)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at org.keycloak.quarkus._private.IDELauncher.main(IDELauncher.java:57)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devui.deployment.BuildTimeContentProcessor#createIndexHtmlTemplate threw an exception: java.lang.NoClassDefFoundError: io/mvnpm/importmap/Aggregator
	at io.quarkus.devui.deployment.BuildTimeContentProcessor.createIndexHtmlTemplate(BuildTimeContentProcessor.java:344)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: java.lang.ClassNotFoundException: io.mvnpm.importmap.Aggregator
	at io.quarkus.launcher.RuntimeLaunchClassLoader.findClass(RuntimeLaunchClassLoader.java:25)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:533)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:481)
	... 10 more

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:350)
	... 16 more
Caused by: java.lang.NoClassDefFoundError: io/mvnpm/importmap/Aggregator
	at io.quarkus.devui.deployment.BuildTimeContentProcessor.createIndexHtmlTemplate(BuildTimeContentProcessor.java:344)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: java.lang.ClassNotFoundException: io.mvnpm.importmap.Aggregator
	at io.quarkus.launcher.RuntimeLaunchClassLoader.findClass(RuntimeLaunchClassLoader.java:25)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:533)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:481)
	... 10 more
```

### How to Reproduce?

Run either of the following:
* [`IDELauncher`](https://github.com/keycloak/keycloak/blob/36b01cbea093d38c495c9e0f2ef66feb3a3f9e41/quarkus/server/src/main/java/org/keycloak/quarkus/_private/IDELauncher.java) from the IDE
* `mvn -f server/pom.xml compile quarkus:dev -Dquarkus.args="start-dev"`

### Anything else?

_No response_

--- a/quarkus/server/pom.xml
+++ b/quarkus/server/pom.xml
@@ -40,6 +40,11 @@
             <artifactId>ojdbc11</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>io.mvnpm</groupId>
+            <artifactId>importmap</artifactId>
+            <scope>provided</scope>
+        </dependency>
     </dependencies>
 
     <properties>

